### PR TITLE
Refs [TASK][26331] Json parse

### PR DESCRIPTION
**Pull request by toandv98**
## Related Tickets

- [#TASK 26331](https://edu-redmine.sun-asterisk.vn/issues/26331)

## WHAT

- Create data class for Api response, create function parse response to data class.

## HOW

- Create data class from Json file, create extension function parse json string to List.

## Review Checklist

| Category      | View Point                                                   | Description                                                  | Expected Reviewer Answer | Reviewer1 (ToanDV) | Reviewer2 (name)   |
| ------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------ | ------------------ | ------------------ |
| General       | Code should be self-explanatory.                             | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Is this function or class too big? If yes, is the function or class having too many responsibilities? |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| General       | Does the code follow DRY?: Do not Repeat Yourself            | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Exception handling and cleanup (dispose) resources           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Does the code build and execute correctly?                   |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Conventions   | Does the source code have end-line errors?                   |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| Conventions   | No hard coding, use constants/configuration values           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Conventions   | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Redmine       | Does the ticket follow Sun* Redmine working process?         | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |

## Notes (Optional)

*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/edu/sun/oereminder/utils/JsonExtension.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    }
+}
+
+@Throws(JSONException::class)
+fun JSONObject.toAccount() = Account(
+    getInt(ACCOUNT_ID),
+    getString(AVATAR_IMAGE_URL),
+    getString(LOGIN_MAIL),
+    getString(NAME)
+)
```
Comment (datdt-1935) [inline app/src/main/java/com/edu/sun/oereminder/utils/JsonExtension.kt]: những cái này nên tạo constructor của từng class sẽ tốt hơn

Comment (toandv98) [reply app/src/main/java/com/edu/sun/oereminder/utils/JsonExtension.kt]: Em Throw JSONException ở đây để không phải Throw Exception từ constructor anh ạ.

Comment (huytq-2422) [reply app/src/main/java/com/edu/sun/oereminder/utils/JsonExtension.kt]: Thêm 1 constructor trong class Account 
```suggestion
Account(jsonObject)
```

File: app/src/main/java/com/edu/sun/oereminder/data/model/Report.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+                "[title]1. Plan:[/title]\n" +
+                "$plan\n" +
+                "[title]2. Actual:[/title]\n" +
+                "$actual\n" +
+                "[title]3. Next:[/title]\n" +
+                "$next\n" +
+                "[title]4. Issue:[/title]\n" +
+                "$issue\n" +
+                "[/info]"
+    }
```
Comment (tuanpa-0908) [inline app/src/main/java/com/edu/sun/oereminder/data/model/Report.kt]: t suggest thay vì override default method của class, ô có thể dùng biến kotlin như sau và dùng như thuộc tính tường minh hơn:
private val content get() = "..."
khi dùng với thể hiện report thì gọi report.content là xong .
String kia cũng nên tránh hard code phòng thay đổi sau này nha

**Review by huytq-2422 — APPROVED** [review]

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**